src/renderer/renderer.js

```diff
@@ -530,6 +550,62 @@ const HanabiRenderer = GObject.registerClass(
                 this._media.pause();
         }
 
+        setAutoWallpaper() {
+            let currentIndex = 0; // Index to keep track of the current video
+
+            let changeWallpaperDirectoryPath = extSettings.get_string('change-wallpaper-directory-path');
+            let videoPaths = [];
+            let dir = Gio.File.new_for_path(changeWallpaperDirectoryPath);
+            let enumerator = dir.enumerate_children(
+                'standard::*',
+                Gio.FileQueryInfoFlags.NONE,
+                null
+            );
+
+            // Get files to push into array
+            let fileInfo;
+            while ((fileInfo = enumerator.next_file(null))) {
+                if (fileInfo.get_content_type().startsWith('video/')) {
+                    let file = dir.get_child(fileInfo.get_name());
+                    videoPaths.push(file.get_path());
+                }
+            }
+            
+            videoPaths = videoPaths.sort();
+
+            function getRandomIndex(actualIndex, videosLength) {
+                if (videosLength <= 1)
+                    return actualIndex;
+
+                let newIndex;
+                do {
+                    newIndex = Math.floor(Math.random() * videosLength);
+                } while (newIndex === actualIndex);
+                return newIndex;
+            }
+
+            function mainLoop() {
+                if (videoPaths.length > 0 && changeWallpaper) {
+                    let videoPath = videoPaths[currentIndex];
+                    extSettings.set_string('video-path', videoPath);
```

Comment:
The variable `videoPath` is already defined at the global. Either use another variable name or you could simply put these into one line: `extSettings.set_string('video-path', videoPaths[currentIndex]);`